akka-actor/src/main/scala/akka/actor/ActorPath.scala

```diff
@@ -272,11 +272,11 @@ final case class RootActorPath(address: Address, name: String = "/") extends Act
 
   override val toSerializationFormat: String = toString
 
-  override def toStringWithAddress(addr: Address): String =
+  override def toStringWithAddress(newAddress: Address): String =
```

Comment:
Yeah this is breaking only for scala in that specific use case. 
Scala has tools to deal with that though - the `def method(@deprecatedName("addr") newOne: Address)` syntax can make that work, though I'm unsure we care enough about this name here to make the change - let's not change it in method params perhaps?